src/managers/input/InputManager.cpp

```diff
@@ -253,6 +253,24 @@ void CInputManager::onMouseButton(wlr_pointer_button_event* e) {
         
 }
 
+void CInputManager::onMouseWheel(wlr_pointer_axis_event* e) {
+    const auto PKEYBOARD = wlr_seat_get_keyboard(g_pCompositor->m_sSeat.seat);
+
+    if (wlr_keyboard_get_modifiers(PKEYBOARD) == (uint32_t)g_pConfigManager->getInt("general:main_mod_internal") && 
+        e->source == WLR_AXIS_SOURCE_WHEEL && e->orientation == WLR_AXIS_ORIENTATION_VERTICAL) {
+            
+        if (e->delta < 0) { 
+            g_pKeybindManager->m_mDispatchers["workspace"]("-1");
```

Comment:
why not use the appropriate func?

Reply:
@vaxerski 

> why not use the appropriate func?

Which one do you mean?  `KeybindManager::changeworkspace` is private and I suppose was meant to be used internally as almost everything else in that class.

Why exactly only the map of functions is exposed? While lurking the code I was thinking why there is no class that exposes public functions for wm manipulations + KeybindManager just using them?

Again, my knowledge of c++ is limited, so maybe I get something wrong.

Reply:
no yeah i forgot this is outside keybindmanager. fine then

they are private to not incentivise their usage outside of keybindmanager